### ISSUE TITLE
Call update_info_plist follow by set_info_plist_value result in bplist plist file

### ISSUE THREAD

**Issue by sainttail** [fastlane/fastlane#6810]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

When I run actions 

```
update_info_plist(
      display_name: 'new name',
      plist_path: plist_path,
      xcodeproj: xcodeproj
    )
set_info_plist_value(
    key: 'CFBundleVersion',
    value: '2222',
    path: params[:plist_path]
  )
```
Somehow it changes normal xml `plist` to `bplist` and causes the consequence action that related to reading `plist` file broken

If I either run only one of the above action or run `set_info_plist_value` first, the `plist` is update correctly.

### You can find the test project [here](https://github.com/sainttail/TestFastlane)

I don't know this is related or not but on my machine I have 2 versions of xcode (7.3.1 and 8.0)

[INSERT OUTPUT HERE]

Here is the output everything look normal. However, if I add a action `set_build_number(plist_path: plist_path)` after plist become bplist file. It will show `invalid byte sequence in UTF-8`

```
~
~ ❯❯❯ cd De...
~ ❯❯❯ cd Desktop/per...
~ ❯❯❯ cd Desktop/personal-project/Test...
~ ❯❯❯ cd Desktop/personal-project/TestFastlane/
~ ❯❯❯ cd Desktop/personal-project/TestFastlane
~ ❯❯❯ cd Desktop/personal-project/TestFastlane
~/D/p/TestFastlane git:master ❯❯❯ fastlane beta                               ✱
[09:27:22]: ------------------------------
[09:27:22]: --- Step: default_platform ---
[09:27:22]: ------------------------------
[09:27:22]: ---------------------------
[09:27:22]: --- Step: opt_out_usage ---
[09:27:22]: ---------------------------
[09:27:22]: Disabled upload of used actions
[09:27:22]: Driving the lane 'ios beta' 🚀
[09:27:22]: -------------------------------
[09:27:22]: --- Step: update_info_plist ---
[09:27:22]: -------------------------------
[09:27:22]: Updated TestFastlane/Info.plist 💾.
[09:27:22]: ----------------------------------
[09:27:22]: --- Step: set_info_plist_value ---
[09:27:22]: ----------------------------------

+------+----------------------+-------------+
|             fastlane summary              |
+------+----------------------+-------------+
| Step | Action               | Time (in s) |
+------+----------------------+-------------+
| 1    | default_platform     | 0           |
| 2    | opt_out_usage        | 0           |
| 3    | update_info_plist    | 0           |
| 4    | set_info_plist_value | 0           |
+------+----------------------+-------------+

[09:27:22]: fastlane.tools finished successfully 🎉
~/D/p/TestFastlane git:master ❯❯❯
```

Here the gif to show how it happens

![test](https://cloud.githubusercontent.com/assets/3015139/19954221/1a7706ee-a1a9-11e6-8e5b-c30a908fb36d.gif)

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.1                                     |
| Ruby                | 2.3.1                                       |
| Bundler?            | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 7.3.1                                       |
| Git                 | git version 2.7.4 (Apple Git-66)            |
| Installation Source | /usr/local/bin/fastlane                     |
| Host                | Mac OS X 10.12.1 (16B2555)                  |
| Ruby Lib Dir        | /usr/local/Cellar/ruby/2.3.1/lib            |
| OpenSSL Version     | OpenSSL 1.0.2g  1 Mar 2016                  |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
#!/usr/bin/ruby

default_platform :ios

opt_out_usage
skip_docs

def set_build_number(params)

  set_info_plist_value(
    key: 'CFBundleVersion',
    value: '2222',
    path: params[:plist_path]
  )
end

platform :ios do

  plist_path = 'TestFastlane/Info.plist'
  xcodeproj  = 'TestFastlane.xcodeproj'

  desc "Builds a beta ipa and uploads it to internal app store and crashlytics"
  lane :beta do |options|
  
    update_info_plist(
      display_name: 'new name',
      plist_path: plist_path,
      xcodeproj: xcodeproj
    )

    set_build_number(plist_path: plist_path)
  end
end
```
</details>

**No Appfile found**

### fastlane gems

| Gem                 | Version | Update-Status |
| ------------------- | ------- | ------------- |
| credentials_manager | 0.16.2  | ✅ Up-To-Date  |
| spaceship           | 0.36.2  | ✅ Up-To-Date  |
| fastlane_core       | 0.53.0  | ✅ Up-To-Date  |
| deliver             | 1.14.5  | ✅ Up-To-Date  |
| snapshot            | 1.16.3  | ✅ Up-To-Date  |
| frameit             | 3.0.0   | ✅ Up-To-Date  |
| pem                 | 1.3.2   | ✅ Up-To-Date  |
| cert                | 1.4.3   | ✅ Up-To-Date  |
| sigh                | 1.11.2  | ✅ Up-To-Date  |
| produce             | 1.2.1   | ✅ Up-To-Date  |
| gym                 | 1.11.3  | ✅ Up-To-Date  |
| pilot               | 1.11.1  | ✅ Up-To-Date  |
| scan                | 0.13.1  | ✅ Up-To-Date  |
| supply              | 0.7.1   | ✅ Up-To-Date  |
| match               | 0.10.0  | ✅ Up-To-Date  |
| screengrab          | 0.5.5   | ✅ Up-To-Date  |
| fastlane            | 1.106.2 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded***

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.2        |
| highline                  | 1.7.8        |
| commander                 | 4.4.0        |
| terminal-table            | 1.4.5        |
| json                      | 1.8.3        |
| multipart-post            | 2.0.0        |
| faraday                   | 0.9.2        |
| faraday_middleware        | 0.10.0       |
| dotenv                    | 2.1.1        |
| net-ssh                   | 3.2.0        |
| net-sftp                  | 2.1.2        |
| plist                     | 3.1.0        |
| security                  | 0.1.3        |
| krausefx-shenzhen         | 0.14.10      |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.3        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.1.2        |
| xcodeproj                 | 1.4.0        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| addressable               | 2.4.0        |
| credentials_manager       | 0.16.2       |
| multi_xml                 | 0.5.5        |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161021 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 1.6.8        |
| xcode-install             | 2.0.8        |
| word_wrap                 | 1.0.0        |
| multi_json                | 1.12.1       |
| babosa                    | 1.0.2        |
| excon                     | 0.54.0       |
| rubyzip                   | 1.1.7        |
| gh_inspector              | 1.0.2        |
| fastlane_core             | 0.53.0       |
| bundler                   | 1.13.3       |
| mini_magick               | 4.5.1        |
| xcpretty-travis-formatter | 0.0.4        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.2.4      |
| google-api-client         | 0.9.19       |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| activesupport             | 4.2.7.1      |
| io-console                | 0.4.6        |
</details>

*generated on:* **2016-11-01**
</details>

**Comment by TKBurner** [comment]
Hi @sainttail Can you share the full output when this happens? I want to take a look.

**Comment by sainttail** [comment]
@TKBurner I update the description and include the gif file thanks.

**Comment by TKBurner** [comment]
@sainttail Thanks for following up. Can you let me know if this happens on a blank Xcode project? If it does, if you can provide exact repro steps that'll help. I want to send this out to the core team. Thanks!

**Comment by kpogz** [comment]
i think i am getting the same error:

/Users/scc/.gem/gems/xcodeproj-1.4.1/lib/xcodeproj/plist.rb:84:in `match': [!] invalid byte sequence in UTF-8 (ArgumentError)
       	from /Users/scc/.gem/gems/xcodeproj-1.4.1/lib/xcodeproj/plist.rb:84:in `match'
       	from /Users/scc/.gem/gems/xcodeproj-1.4.1/lib/xcodeproj/plist.rb:84:in `file_in_conflict?'
       	from /Users/scc/.gem/gems/xcodeproj-1.4.1/lib/xcodeproj/plist.rb:20:in `read_from_path'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/actions/update_info_plist.rb:37:in `run'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:202:in `block (2 levels) in execute_action'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/actions/actions_helper.rb:35:in `execute_action'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:187:in `block in execute_action'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:186:in `chdir'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:186:in `execute_action'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:112:in `trigger_action_by_name'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/fast_file.rb:140:in `method_missing'
       	from Fastfile:113:in `block in parsing_binding'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/lane.rb:33:in `call'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/lane.rb:33:in `call'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:165:in `try_switch_to_lane'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:120:in `trigger_action_by_name'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/fast_file.rb:140:in `method_missing'
       	from Fastfile:94:in `block in parsing_binding'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/lane.rb:33:in `call'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/lane.rb:33:in `call'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:49:in `block in execute'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:45:in `chdir'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/runner.rb:45:in `execute'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/command_line_handler.rb:30:in `handle'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/commands_generator.rb:69:in `block (2 levels) in run'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/commander-4.4.0/lib/commander/command.rb:178:in `call'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/commander-4.4.0/lib/commander/command.rb:178:in `call'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/commander-4.4.0/lib/commander/command.rb:153:in `run'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/commander-4.4.0/lib/commander/runner.rb:444:in `run_active_command'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane_core-0.53.0/lib/fastlane_core/ui/fastlane_runner.rb:38:in `run!'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/commander-4.4.0/lib/commander/delegates.rb:15:in `run!'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/commands_generator.rb:257:in `run'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/commands_generator.rb:31:in `start'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/lib/fastlane/cli_tools_distributor.rb:58:in `take_off'
       	from /Users/scc/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/fastlane-1.107.0/bin/fastlane:5:in `<top (required)>'
       	from /Users/scc/.rbenv/versions/2.2.2/bin/fastlane:22:in `load'
       	from /Users/scc/.rbenv/versions/2.2.2/bin/fastlane:22:in `<main>'

let me know if youd like more info

**Comment by sainttail** [comment]
@TKBurner I've already attached the repo in the comment section lol.

[Here](https://github.com/sainttail/TestFastlane) is it

**Comment by sainttail** [comment]
@TKBurner I think it happens with the blank repo [here](https://github.com/sainttail/TestFastlane).

For this repo,
1. Create a blank project
2. Add `fastfile` which have that two actions
3. Run `fastlane lane`
4. Plist change!

PS. Seem like this will not happen on my colleagues machine. Look like it is the problem of combination of ruby gem or ruby version.

**Comment by kpogz** [comment]
Is this still being looked into? I've been stuck on this for awhile, my fastlane deployment is halted :( 
sorry if this is kinda dumb but what does the 'action' label mean?
let me know if theres anything i can do to help!

**Comment by kpogz** [comment]
my coworked an i solved the issue.
we had to change the order of the update_info_plist,  update_app_identifier, set_info_plist_value functions. there may be a race condition or something!
woot!

**Comment by mfurtak** [comment]
@sainttail Thanks for the really excellent bug report 👍  I can definitely reproduce the behavior with your test project, and see that your PR seems to prevent the problem.

I can't, however, explain why this is so. 😄  Since `to_plist` just calls `Plist::Emit.dump` internally, they should be equivalent.

Since they _should_ be equivalent, but your adjustment fixes things, I think the change is worth taking.

**Comment by mfurtak** [comment]
Merged #7188 - that will go out in the next _fastlane_ release this week.
